### PR TITLE
NPC Fishing, Chopping logs and trees, and NPC butchery

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "NPC Fishing, Chopping logs and trees, and NPC butchery"```

#### Purpose of change
As per the previous 4 PRs in this series. ( see linked PR below )

this relies upon 
#32730 

and therefore is marked as WIP until they are merged.

This adds a new activity for NPCs,
There is a new zone, in zone manager - Fishing spot, designate this over deep water ( allowing shalow water nearby for NPC to stand in and fish into the deep water )
When NPC starts the activity, they will look for a fishing rod, go and pick it up, then go to the nearest fishing spot, and fish for 5 hours., when they are done, any fish they catch, they will sort to the nearest unsorted loot zone, along with the fishing rod they picked up.

#### Describe the solution
Same as the other new NPC activity loops

#### Describe alternatives you've considered
N/A

#### Additional context
EDIT - Have closed the previous 3 PRs for chopping logs, butchering, chopping trees, as this one contains those changes anyway, and this one has latest fixes for the generic activity loop, which ive rebased #32730 anyway

### CODE REVIEW

**Comment by davidpwbrown** [comment]
jenkins rebuild

**Comment by davidpwbrown** [comment]
jenkins rebuild

**Comment by davidpwbrown** [comment]
jenkins rebuild

**Comment by cainiaowu** [comment]
You may need to update the summary line as well.

**Comment by davidpwbrown** [comment]
> You may need to update the summary line as well.

Yup. thanks

File: src/activity_handlers.h [hunk 1]
```diff
@@ -52,7 +52,7 @@ enum do_activity_reason : int {
     BLOCKING_TILE           // Something has made it's way onto the tile, so the activity cannot proceed
 };
 
-int butcher_time_to_cut( const player &u, const item &corpse_item, const butcher_type action );
```
Comment (BevapDin) [inline src/activity_handlers.h]: Why?

File: src/activity_item_handling.cpp [hunk 2]
```diff
@@ -1084,7 +1084,7 @@ static std::pair<bool, do_activity_reason> can_do_activity_there( const activity
                 b_rack_present = true;
             }
         }
-        if( corpses.size() > 0 ) {
+        if( corpses.empty() ) {
```
Comment (BevapDin) [inline src/activity_item_handling.cpp]: This inverts the meaning of the condition. And it makes the checks for `big_count > 0` below always false, so it does not seem to be a correct change.

File: src/activity_handlers.h (hunk 1 above)
Comment (davidpwbrown) [reply src/activity_handlers.h]: hmm, no idea. thats weird. obvious mistake. will restore.

Comment (davidpwbrown) [reply src/activity_handlers.h]: ah thats due to clang-tidy aswell, about the const parameter.

**Comment by davidpwbrown** [comment]
Ok so ive rebased but im only 50% sure it now fits into the reworked functions that #33673 introduced, will test more tomorrow.

**Comment by davidpwbrown** [comment]
Seems to work fine after the messy rebase.

**Pull request merged**